django/thunderstore/repository/api/v1/views/metrics.py

```diff
@@ -0,0 +1,54 @@
+from drf_yasg.utils import swagger_auto_schema
+from rest_framework import serializers
+from rest_framework.generics import get_object_or_404
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from thunderstore.repository.models import Package, PackageVersion
+
+
+class PackageMetricsSerializer(serializers.Serializer):
+    downloads = serializers.IntegerField()
+    rating_score = serializers.IntegerField()
+
+
+# This view is potentialy used by the shields.io project and should be kept
+# backwards-compatible
+class PackageMetricsApiView(APIView):
+    @swagger_auto_schema(
+        operation_id="api_v1_package_metrics",
+        responses={200: PackageMetricsSerializer()},
+        tags=["v1"],
+    )
+    def get(self, request, namespace: str, name: str):
```

Comment:
I find partly type hinted methods a bit odd. (Also in the other view.)

Reply:
Couldn't tell you why I chose to not type requests anymore, but clearly a habit from somewhere. If I had to guess it's because some views rely on middleware-provided request parameters, which isn't possible to type correctly AFAIK.